torch/distributed/rpc/api.py

```diff
@@ -147,6 +207,10 @@ def _init_rpc_backend(
     )
     _start_rpc_agent(_agent)
 
+    worker_infos = _agent.get_worker_infos()
+    global _ALL_WORKER_NAMES
+    _ALL_WORKER_NAMES = {worker_info.name for worker_info in worker_infos}
```

Comment:
Can you clarify the intent of caching the worker names as a global here ? We can query the agent whenever we need it right ?

Reply:
It's for `def _on_leader_follower_report_intent(worker_name):` to check validity.